src/editor/objects/Model.js

```diff
@@ -63,29 +64,58 @@ export default class Model extends Object3D {
       this.model && this.model.animations
         ? this.model.animations.map((clip, index) => ({ label: clip.name, value: index }))
         : [];
-    clipOptions.unshift({ label: "None", value: -1 });
+    if (clipOptions.length == 0) {
+      clipOptions.unshift({ label: "None", value: -1 });
+    }
     return clipOptions;
   }
 
-  get activeClip() {
-    return (this.model && this.model.animations && this.model.animations[this.activeClipIndex]) || null;
+  get activeClipNames() {
+    return this.activeClipIndices.filter(item => item.value >= 0).map(item => item.label);
+  }
+
+  getActiveClipIndices() {
```

Comment:
Should `getActiveClipIndices` return `this.activeClipIndices`? A bit confused on the naming / control flow here.

Reply:
Actually I've seen a bug reviewing this. We are not correctly restoring the animations, we are updating the `activeClipIndices` with the loop-animation stored indexes when deserializing. We need to convert that into a label/value struct so we can feed it to the React component. I'll address in the follow-up PR.